pkg/kubernetes/namespace/v1alpha1/namespace.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright © 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	errors "github.com/openebs/maya/pkg/errors/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Namespace is a wrapper over Namespace api
+// object. It provides build, validations and other common
+// logic to be used by various feature specific callers.
+type Namespace struct {
+	object *corev1.Namespace
+}
+
+// Builder enables building an instance of StorageClass
+type Builder struct {
+	ns   *Namespace
+	errs []error
+}
+
+// NewBuilder returns new instance of Builder
+func NewBuilder() *Builder {
+	return &Builder{ns: &Namespace{object: &corev1.Namespace{}}}
+}
+
+// WithName sets the Name field of namespace with provided argument.
+func (b *Builder) WithName(name string) *Builder {
+	if len(name) == 0 {
+		b.errs = append(b.errs, errors.New("failed to build storageclass: missing storageclass name"))
+		return b
+	}
+	b.ns.object.Name = name
+	return b
+}
+
+// Build returns the namespace API instance
+func (b *Builder) Build() (*corev1.Namespace, error) {
```

Comment:
`Build` should not return API object type.
Think of wrapper objects as the internal custom object with whom you want to associate behaviours.
It should return `*Namespace` type.

Reply:
I will agree with that. 
What would you say 
cc: @AmitKumarDas

Reply:
Done